src/SFA.DAS.Roatp.CourseManagement.Web/Startup.cs

```diff
@@ -1,49 +1,142 @@
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure.Authorization;
 
 namespace SFA.DAS.Roatp.CourseManagement.Web
 {
     [ExcludeFromCodeCoverage]
     public class Startup
     {
+        private readonly IConfigurationRoot _configuration;
+
         public Startup(IConfiguration configuration)
         {
             var config = new ConfigurationBuilder()
                 .AddConfiguration(configuration)
-                .SetBasePath(Directory.GetCurrentDirectory())
-                .AddEnvironmentVariables();
+                .SetBasePath(Directory.GetCurrentDirectory());
+#if DEBUG
+            if (!configuration["EnvironmentName"].Equals("DEV", StringComparison.CurrentCultureIgnoreCase))
+            {
+                config.AddJsonFile("appsettings.json", true)
+                    .AddJsonFile("appsettings.Development.json", true);
+            }
+#endif
+            config.AddEnvironmentVariables();
 
-            config.AddAzureTableStorage(options =>
+            if (!configuration["EnvironmentName"].Equals("DEV", StringComparison.CurrentCultureIgnoreCase))
             {
-                options.ConfigurationKeys = configuration["ConfigNames"].Split(",");
-                options.StorageConnectionString = configuration["ConfigurationStorageConnectionString"];
-                options.EnvironmentName = configuration["Environment"];
-                options.PreFixConfigurationKeys = false;
-            });
+                config.AddAzureTableStorage(options =>
+                {
+                    options.ConfigurationKeys = configuration["ConfigNames"].Split(",");
+                    options.StorageConnectionString = configuration["ConfigurationStorageConnectionString"];
+                    options.EnvironmentName = configuration["EnvironmentName"];
+                    options.PreFixConfigurationKeys = false;
+                }
+                );
+            }
 
-            Configuration = config.Build();
+            _configuration = config.Build();
         }
 
-        public IConfiguration Configuration { get; }
-
-        // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddControllersWithViews();
-            services.AddProviderUiServiceRegistration(Configuration);
+            services.Configure<CookiePolicyOptions>(options =>
+            {
+                options.CheckConsentNeeded = context => true;
+                options.MinimumSameSitePolicy = SameSiteMode.None;
+            });
+
+            services.AddConfigurationOptions(_configuration);
+
+            services.AddSingleton<IAuthorizationHandler, ProviderAuthorizationHandler>();
+
+            services.AddProviderUiServiceRegistration(_configuration);
+
+            /// services.AddMediatR(typeof(GetCreateCourseDemandQuery).Assembly);
+            /// services.AddMediatRValidation();
+
```

Comment:
Are these coming back in at some point? If so, I'm fine with leaving them in, commented out. Otherwise remove

Reply:
yes, that is why I left it in there.